src/Microsoft.IdentityModel.Tokens/TokenValidationParameters.cs

```diff
@@ -240,7 +240,13 @@ public virtual ClaimsIdentity CreateClaimsIdentity(SecurityToken securityToken,
             if (LogHelper.IsEnabled(EventLogLevel.Informational))
                 LogHelper.LogInformation(LogMessages.IDX10245, securityToken);
 
-            return new ClaimsIdentity(authenticationType: AuthenticationType ?? DefaultAuthenticationType, nameType: nameClaimType ?? ClaimsIdentity.DefaultNameClaimType, roleType: roleClaimType ?? ClaimsIdentity.DefaultRoleClaimType);
+            if (AppContextSwitches.UseClaimsIdentityType())
```

Comment:
I think all this can be moved to the factory. Basically this switch check should only have one usage in prod code: in the factory.